### DISCUSSION TITLE
Support for virtual fields from config

### DISCUSSION THREAD

**Discussion by r1tsuu** [payloadcms/payload#6270] [Feature Requests & Ideas]
There's a good blog post about Virtual Fields https://payloadcms.com/blog/learn-how-virtual-fields-can-help-solve-common-cms-challenges

And yes, it's true, they are great. But if you are using SQL database, the schema for this field is still created, because there's no way to determine whatever field is "virtual" or not.

My proposal would be to add a field option like:
`isVirtual: true` or `db: { store: false }` so then in our adapter we can skip the schema for this field, but instead we should validate for `afterRead` hook existance and automatically delete this field before `create` / `update` operation.

PR: https://github.com/payloadcms/payload/pull/7621

**Comment by rainer-friederich** [comment]
We just stumbled upon this. This would really be a required thing for a more sophisticated site.

**Comment by r1tsuu** [comment]
I can create a PR on this, the only what I'm thinking about is an interface for that (whatever prop `isVirtual` or something) and how it should affect hooks, or shouldn't at all

**Reply by rainer-friederich** [reply]
I think we should perhaps separate the schema and the store mechanism to not influnce the current behaviours and hooks?
Something like:
```
db: { 
  store: boolean,
  table: boolean
}
```

**Comment by rainer-friederich** [comment]
I think also payload core has a use case for this with the lexical HTML feature which saved html inside the database.
https://github.com/payloadcms/payload/pull/6795

**Comment by r1tsuu** [comment]
PR: https://github.com/payloadcms/payload/pull/6983

**Comment by r1tsuu** [comment]
Released iin https://github.com/payloadcms/payload/releases/tag/v3.0.0-beta.108
